proposed/0000-Farm-attempt.md

```diff
@@ -0,0 +1,56 @@
+- Feature Name:Farm attempt 
+- Type new feature
+- Related components maidsafe_vault
+- Start Date: 24-06-2015 
+- RFC PR:
+- Issue number: 
+
+# Summary
+
+This proposal outlines an initial design to test the efficiency and security of farming attempts.  This involves making use of the Farming Rate (FR) which is the rate at which farming attempts are made.  It is assumed the farming rate will include payments to care development, content producers and most significantly farmers (or providers of the network resource). 
+
+# Motivation
+
+To ensure network stability and resource availability a reward mechanism is required to ensure the use of resource is balanced with the supply of resource. 
+
+# Scope of work
+
+This implementation will be solely in the vault library. This will include several steps. The client lib will require to be aware of the actual safecoin type (again `StructuredData`). 
+
+# Detailed design
+
+ON receipt of a `Get` request for `ImmutableData` the `DataManager` calculates the SHA512 Hash of the `name()` of the `ImmutableData` concatenated with the
+`PmidHolder`. This result is then modulo divided by the farming rate (FR). The rate attempts should divide the award between 4 distinct groups. 
+
+##Rmte attempts 
+
+1. PmidNodes -> tested at 100% of FR [i.e. Farming rate * 1]
+2. App Developer -> tested at 10% of FR [i.e. Farming rate * 1.1] 
+3. Publisher -> tested at 10% of FR [i.e. Farming rate * 1.1]
+4. Core development -> tested at 5% of FR [i.e. Farming rate * 1.05]
+
+##Payment address 
+
+1. PmidNodes -> Registers an Optional (if set by user) wallet address on any store of data to it
+2. App Developer -> App developers will include wallet address in any `Get` request
+3. Publisher -> As data is stored a wallet address of the owner is stored if this is first time seen on the network (stored in DM account for the data element)
+4. Core development -> Initially every node will be aware of a hard coded wallet address for core development. This will likely lead to a multisig wallet.
+
+##Farm process
+
+When the initial farming test above is true (the modulo division ==0 ) then the `DataManagers` send a Farm request message `Post` to the group closest to the
+combined Hash (i.e. Hash of (`PmidHolder` + `ImmutableData` )). This request includes the original hashes + hash of chunk requested +current farming rate of the group as well as wallet address. This is confirmed at the receiving group and they check for the existence of a safecoin space (ie, there is no safecoin with this name). If this is successful then a further check is made to confirm there is a `DataManager` group (implicit as routing does this). A safecoin is then created and a message sent to the wallet holder of the request.  All these requests must be digitally signed and confirmed at receiving end (implicit in sentnel). 
```

Comment:
Typo: 
   (i.e. Hash of (`PmidHolder` + `ImmutableData` ))
Order should be same as explained in detail design section.
   `DataManager` calculates the SHA512 Hash of the `name()` of the `ImmutableData` concatenated with the
   `PmidHolder`